Invalid initialization of char* with a single char

I see the following error when building with GCC 7:

````
/builddir/build/BUILD/percolator-rel-3-01/src/converters/MSToolkit/MSReader.h:85:80: error: invalid conversion from 'char' to 'char*' [-fpermissive]
   void writeFile(const char* c, MSFileFormat ff, MSObject& m, char* sha1Report='\0');
                                                                                ^~~~
````

What is this trying to do? You can't assign a single `char` to a `char*`. In older versions of C++ this would be the same as `char* sha1Report = 0` i.e. create a null pointer. In C++ today it's an error.

Is the code attempting to create an empty string, i.e. `char* sha1Report = ""` ?

That won't work, because it would need to be `const char* sha1Report = ""` (which it probably should be anyway, since you don't modify the string, but that would be an API and ABI change to the library).

Or is it trying to create a null pointer, i.e. `char* sha1Report = 0`?

That will compile, but will then pass a null pointer to `sprintf` in `MSReader::writeSqlite` which surely can't be what you intended.

--- a/src/converters/MSToolkit/MSReader.cpp
+++ b/src/converters/MSToolkit/MSReader.cpp
@@ -585,6 +585,9 @@ void MSReader::writeFile(const char* c, MSFileFormat ff, MSObject& m, char* sha1
 #ifndef _NOSQLITE
 void MSReader::writeSqlite(const char* c, MSObject& m, char* sha1Report)
 {
+  char empty[1] = "";
+  if(!sha1Report)
+    sha1Report = empty;
 
   //open the database for write
   sqlite3_open(c, &db);

--- a/src/converters/MSToolkit/MSReader.h
+++ b/src/converters/MSToolkit/MSReader.h
@@ -82,7 +82,7 @@ class MSReader {
   void setPrecisionInt(int i);
   void setPrecisionMZ(int i);
   void writeFile(const char* c, bool text, MSObject& m);
-  void writeFile(const char* c, MSFileFormat ff, MSObject& m, char* sha1Report='\0');
+  void writeFile(const char* c, MSFileFormat ff, MSObject& m, char* sha1Report=0);
 
   bool readFile(const char* c, bool text, Spectrum& s, int scNum=0);
   bool readFile(const char* c, MSFileFormat f, Spectrum& s, int scNum=0);

